### PR TITLE
Adding DataCollection parameter, updating test parameters, and refining module dependencies.

### PR DESCRIPTION
**Pull request by kwirkykat**
These changes include Adding DataCollection parameter, updating test parameters, and refining module dependencies.

Required modules in the module manifest have been refined from all AzureRm to just AzureRm.Profile and AzureRm.Automation.

The KeyPair parameter has been removed from tests since all new instances should be removed by tests automatically.
The ExtensionVersion parameter has been added to tests to allow testing new versions of the 'AWS DSC Extension'.

The DataCollection parameter has been added to Register-EC2Instance.
This parameter allows the user to opt-out of sending telemetry data about the 'AWS DSC Extension' to Microsoft.

Telemetry data collection by Microsoft for the 'AWS DSC Extension' is currently disabled for all AWS machines.

The only valid values for the DataCollection parameter are 'Enable' and 'Disable'.

Not specifying the DataCollection parameter or supplying the 'Enable' value will allow Microsoft to collect telemetry data under this [privacy agreement](http://go.microsoft.com/fwlink/p/?linkid=131004&amp;clcid=0x409) which has been added to the README.

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/awsdsctoolkit/3)

### CODE REVIEW

File: AwsDscToolkit.psm1 [hunk 1]
```diff
@@ -922,6 +1004,10 @@ function Register-EC2Instance {
         [string]
         $AwsEncryptionKeyId,
 
+        [ValidateSet('Enable', 'Disable')]
+        [string]
```
Comment (TravisEz13) [inline AwsDscToolkit.psm1:1008]: Will you be explicitly passing null to this?

File: AwsDscToolkit.psm1 [hunk 2]
```diff
@@ -1520,15 +1645,25 @@ function Test-EC2InstanceRegistration {
...
+
+        [string]
+        $AzureAutomationResourceGroup,
 
         # AWS info
-        [string]$AwsRegion,
-        [string]$AwsAccessKey,
-        [string]$AwsSecretKey,
+        [string]
+        $AwsRegion,
```
Comment (TravisEz13) [inline AwsDscToolkit.psm1:1656]: try not to combine format changes and logic changes in the same review

**Comment by TravisEz13** [comment]
> 

Don't use UTF 16 on github

---

Refers to: AwsDscToolkit.psd1:2 in 2f4c141. [](commit_id = 2f4c141dc033478a525c8b25c4cf663c9525d5ec, deletion_comment = False)

File: Tests/AwsDscToolkit.Tests.ps1 [hunk 3]
```diff
@@ -387,10 +396,12 @@ function Invoke-WaitForEC2InstanceStatus {
...
     # Wait until the instance has the desired status
     while ($instanceStatus.Status.Status.Value -ne $DesiredStatus) {
-        Write-Verbose "$(Get-Date) Instance status is $($instanceStatus.Status.Status.Value). Waiting for 10 seconds..."
-        Start-Sleep -Seconds 10
+        Write-Verbose "$(Get-Date) Instance status is $($instanceStatus.Status.Status.Value). Waiting for 30 seconds..."
+        Start-Sleep -Seconds 30
         $instanceStatus = AWSPowershell\Get-EC2InstanceStatus $InstanceId -AccessKey $AwsAccessKey -SecretKey $AwsSecretKey -Region $AwsRegion
     }
+
+    Write-Verbose "$(Get-Date) Instance status is $($instanceStatus.Status.Status.Value)."
```
Comment (TravisEz13) [inline Tests/AwsDscToolkit.Tests.ps1]: consider a helper function for writing verbose in this format

File: Tests/AwsDscToolkit.Tests.ps1 [hunk 4]
```diff
@@ -845,7 +851,7 @@ try {
                         Invoke-WaitForEC2InstanceState -InstanceId $instanceId -DesiredState 'running' -AccessKey $AwsAccessKey -SecretKey $AwsSecretKey -Region $AwsRegion -Verbose
                         Invoke-WaitForEC2InstanceStatus -InstanceId $instanceId -DesiredStatus 'ok' -Verbose
 
-                        $status = Invoke-WaitForEC2InstanceRegistered -InstanceId $instanceId
+                        $status = Invoke-WaitForEC2InstanceRegistered -InstanceId $instanceId -Verbose
```
Comment (TravisEz13) [inline Tests/AwsDscToolkit.Tests.ps1]: Is it a good idea to force verbose?

**Comment by TravisEz13** [comment]
:clock1:

File: AwsDscToolkit.psm1 (hunk 1 above)
Comment (kwirkykat) [inline AwsDscToolkit.psm1:1008]: No. $null is only a valid value for the functions that this function passes to to indicate that the parameter was not supplied initially.

**Comment by TravisEz13** [comment]
Reviewed 1 of 2 files at r1, 3 of 3 files at r7.
Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/awsdsctoolkit/3)_

**Comment by kwirkykat** [comment]
Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

_[Tests/AwsDscToolkit.Tests.ps1, line 404 [r8]](https://reviewable.io:443/reviews/powershell/awsdsctoolkit/3#-KHW5P2gp1LAeqPcf2kX-r8-404:-KHf7ng3SbZMQ6hmQzH0:-499146009) ([raw file](https://github.com/powershell/awsdsctoolkit/blob/2f4c141dc033478a525c8b25c4cf663c9525d5ec/Tests/AwsDscToolkit.Tests.ps1#L404)):_
<details><summary>

<i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>



> consider a helper function for writing verbose in this format</details>
> Done.

---

_[Tests/AwsDscToolkit.Tests.ps1, line 854 [r8]](https://reviewable.io:443/reviews/powershell/awsdsctoolkit/3#-KHW5P2gp1LAeqPcf2kX-r8-854:-KHf7y_VtM8OAlA-gynD:-499146009) ([raw file](https://github.com/powershell/awsdsctoolkit/blob/2f4c141dc033478a525c8b25c4cf663c9525d5ec/Tests/AwsDscToolkit.Tests.ps1#L854)):_
<details><summary>

<i>Previously, TravisEz13 (Travis Plunk) wrote…</i></summary>



> Is it a good idea to force verbose?
> 
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/awsdsctoolkit/3)_

**Comment by TravisEz13** [comment]
Reviewed 1 of 1 files at r8, 2 of 2 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/awsdsctoolkit/3)_

**Pull request merged**